Author: Roger Wesson <rw@nebulousresearch.org>
Description: allow variables to propagate, add install rule, and rm .exe extension from files
--- a/source/Makefile
+++ b/source/Makefile
@@ -3,12 +3,12 @@
 # you need g++ or pathCC to create the dependencies (set with CXXDEPEND)
 # but after that you can use any compiler you want (set with CXX)
 
-OPT = -O3 -ftrapping-math -fno-math-errno 
+OPT = -O3 -ftrapping-math -fno-math-errno
 DEBUGOPT = -finline
 CXX = g++
 # -W is the old name for -Wextra, g++ 3.3 only supports the former
-CXXFLAGS = -ansi ${OPT} -Wall -W -g
-LDFLAGS = ${OPT} -Wall -g
+CXXFLAGS += -ansi ${OPT} -Wall -W -g
+LDFLAGS += ${OPT} -Wall -g
 LDLIBS =
 NO_TRAP_FLAGS =
 # this is intended for supplying extra command line parameters, e.g. extra -D flags
@@ -99,11 +99,9 @@
 
 CLDCONFIG = $(PWD)/cloudyconfig.h
 
-CDP = ${CLOUDY_DATA_PATH}
-ifeq ($(CDP),)
-# create reasonable default when CLOUDY_DATA_PATH not set...
-  CDP = $(PWD)/${SRCDIR}/../data/
-endif
+export CLOUDY_DATA_PATH:=$(PWD)/${SRCDIR}/../data/
+CDP = ${DESTDIR}/usr/share/cloudy/data
+
 ifneq (${CLOUDY_LAPACK_PATH},)
   CXXFLAGS += -DLAPACK -I${CLOUDY_LAPACK_PATH}/include
   LDLIBS += -L${CLOUDY_LAPACK_PATH}/lib -llapack
@@ -139,7 +137,7 @@
 CXXFLAGSNOCFG := ${filter-out ${CFGFILTER},${CXXFLAGS}}
 
 # Makefile.targets can override or add to the DEFAULT variable.
-DEFAULT = cloudy.exe data
+DEFAULT = cloudy data
 
 # include additional targets from file if one exists (or carry on without
 # worrying if it doesn't)
@@ -149,25 +147,25 @@
 
 default: $(DEFAULT)
 
-cloudy.exe: ${preobjects} maincl.${OBJEXT} libcloudy.${LIBEXT}
-	${CXX} ${LDFLAGS} -o cloudy.exe maincl.${OBJEXT} -L. -lcloudy ${LDLIBS}
+cloudy: ${preobjects} maincl.${OBJEXT} libcloudy.${LIBEXT}
+	${CXX} ${LDFLAGS} -o cloudy maincl.${OBJEXT} -L. -lcloudy ${LDLIBS}
 
 libcloudy.${LIBEXT}: ${libobjects}
 	ar ru libcloudy.${LIBEXT} $^
 	${RANLIB} libcloudy.${LIBEXT}
 
 data:
-	cd ${SRCDIR}/../data/cdms+jpl && $(MAKE)
+	cd ${SRCDIR}/../data/cdms+jpl && $(MAKE) && ${MAKE} clean
 
 clean:
 	rm -f *.${OBJEXT}
 	rm -f *.d
 	rm -f *.h.gch
 	rm -rf SunWS_cache
-	rm -rf cloudy.exe.dSYM
+	rm -rf cloudy.dSYM
 	rm -f libcloudy.${LIBEXT}
 	rm -f $(DEFAULT)
-	rm -f runtests.exe
+	rm -f runtests
 	rm -rf tmp_cloudyconfig.*
 
 distclean: clean
@@ -206,13 +204,13 @@
 
 debug-test: test
 
-test: runtests.exe
-	./runtests.exe
+test: runtests
+	./runtests
 
-valgrind-test: runtests.exe
-	valgrind --leak-check=full --track-fds=yes ./runtests.exe
+valgrind-test: runtests
+	valgrind --leak-check=full --track-fds=yes ./runtests
 
-runtests.exe: lib/libUnitTest++.${LIBEXT} libcloudy.${LIBEXT} ${testobjects}
+runtests: lib/libUnitTest++.${LIBEXT} libcloudy.${LIBEXT} ${testobjects}
 	${CXX} ${CXXFLAGS} -${OBJEXT} $@ ${filter-out %.${LIBEXT}, $^} ${LDFLAGS} -L. -lcloudy -Llib -lUnitTest++ ${LDLIBS}
 
 # the dependency on libUnitTest++.a is needed to make sure header files are in place.
@@ -245,3 +243,8 @@
 ifneq (${DEP_GOALS},)
 -include ${predepends} ${fulldepends}
 endif
+
+install:
+	test -e ${DESTDIR}/usr/bin || mkdir -p ${DESTDIR}/usr/bin
+	test -e ${DESTDIR}/usr/share/cloudy || mkdir -p ${DESTDIR}/usr/share/cloudy
+	install cloudy ${DESTDIR}/usr/bin
